packages/manager/src/routes/index.tsx

```diff
@@ -70,6 +70,26 @@ export const router = createRouter({
 declare module '@tanstack/react-router' {
   interface Register {
     // This infers the type of our router and registers it across the entire project
-    router: typeof router;
+    // router: typeof router;
+    migrationRouter: MigrationRouter;
```

Comment:
I believe this is essentially making the router untyped / any. 

I tried updating it restore the type-safety:
```suggestion
    router: MigrationRouter;
```
but this introduces many TypeScript errors because of all of the work we've done for routes that are not yet in the migrationRouter.


Any thoughts here? Will we just have to proceed with an untyped router until the end of this migration?

Maybe we could do something hacky like this, but it might cause us more trouble than we want
```suggestion
    router: typeof router | MigrationRouter;
```

Reply:
@bnussman of course! 🤦 thx for catching.

So look at https://github.com/linode/manager/pull/11049/commits/75b20f0c2ab3f5b9f8bc18912c7a3d7f905536c0

I reintroduced the main router type for the applciation, which also forced me to cast `AnyRouter` in a few places. Still much better! and allowed to get `<Link />` to show path suggestion which is a good idea.

We still have good typing but will not narrow to the `migrationRouter` routes which I was hoping to be able to do. We're half way there and get at least stronger rail guards and working suggestions 👍